.github/workflows/ci.yaml

```diff
@@ -0,0 +1,176 @@
+    runs-on: ubuntu-latest # TODO: until static global env can be used through ${{ env }}.
+    outputs:
+      git_refname: ${{ steps.set-git-refname.outputs.git_refname }}
+      github_runner_default_os: ubuntu-latest
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+
+      - name: Set Git refname
+        id: set-git-refname
+        run: echo ::set-output name=git_refname::$(echo "${{ github.ref }}" | sed -r 's@refs/(heads|pull|tags)/@@g' )
+
+  static-checks:
+    name: Run static code analysis
+    needs:
+      - environment
+    runs-on: ${{ needs.environment.outputs.github_runner_default_os }}
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+
+      - name: Setup Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ env.GO_VERSION }}
+
+      - name: Cache licenses
+        id: cache-licenses
+        uses: actions/cache@v2
+        with:
+          path: .licensei.cache
+          key: licensei-v1-${{ needs.environment.outputs.git_refname }}-${{ hashFiles('go.sum') }}
+          restore-keys: |
+            - licensei-v1-${{ needs.environment.outputs.git_refname }}
+            - licensei-v1-master
+            - licensei-v1
+
+      - name: Download license information for dependencies
+        env:
+          GITHUB_TOKEN: ${{ github.token }} # Note: this is required for licensei authentication to avoid rate-limiting.
+        run: make license-cache
+
+      - name: List license information for dependencies
+        run: ./bin/licensei list
+
+      - name: Check dependency licenses
+        run: make license-check
+
+      - name: Run linter
+        run: make lint
+
+  build:
+    name: Build project
+    needs:
+      - environment
+      - static-checks
+    runs-on: ${{ needs.environment.outputs.github_runner_default_os }}
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+        with:
+          # Required for correctly populating ldflags
+          fetch-depth: 0
+
+      - name: Setup Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ env.GO_VERSION }}
+
+      - name: Cache Go module dependencies
+        id: cache-go-module-dependencies
+        uses: actions/cache@v2
+        with:
+          path: ~/go/pkg/mod
+          key: go-mod-cache-${{ runner.os }}-${{ env.GO_VERSION }}-${{ hashFiles('go.sum') }}
+          restore-keys: |
+            go-mod-cache-${{ runner.os }}-${{ env.GO_VERSION }}
+            go-mod-cache-${{ runner.os }}
+            go-mod-cache
+
+      - name: Build binaries and libraries
+        run: make build
+
+  unit-tests:
+    name: Run unit tests
+    needs:
+      - environment
+      - static-checks
+      - build
+    runs-on: ${{ needs.environment.outputs.github_runner_default_os }}
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+        with:
+          # Required for correctly populating ldflags
```

Comment:
Not that it really matters for tests

Reply:
I wasn't sure whether binaries built for tests require the mentioned ldflags as this comment was a legacy one which I know nothing of.

Fixed in a8ea2e8.